src/librustc_mir/build/matches/mod.rs

```diff
@@ -439,11 +439,17 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
         self.simplify_candidate(&mut candidate);
 
         if !candidate.match_pairs.is_empty() {
-            span_bug!(
+            // Only abort compilation if no other errors have been emitted. This used to be a hard
+            // error that wouldn't be reached because `hair::pattern::check_match::check_match`
+            // wouldn't have let the compiler continue. In our tests this is only ever hit by
+            // `ui/consts/const-match-check.rs` with `--cfg eval1`, and that file already generates
+            // a different error before hand.
```

Comment:
The comment could be changed, but it's not that big of a deal (ftr, `delay_span_bug` is more like "ICE if compilation appears to succeed", i.e. it's cancelled by any errors, not just past ones - which among other things, results in no ordering dependency).